keyboards/troyle/macro_keys_dual/keyboard.json

```
@@ -0,0 +1,44 @@
+{
+    "manufacturer": "troyle",
+    "keyboard_name": "troyle/macro_keys_dual",
+    "maintainer": "troyle",
+    "development_board": "promicro",
+    "encoder": {
+        "rotary": [
+            {"pin_a": "D1", "pin_b": "D0", "resolution": 2},
+            {"pin_a": "B4", "pin_b": "B5", "resolution": 2}
+        ]
+    },
+    "features": {
+        "bootmagic": true,
+        "encoder": true,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true
+    },
+    "matrix_pins": {
+        "direct": [
+            ["F7", "F6", "F5", "F4"],
+            ["B6", "B2", "B3", "B1"]
+        ]
+    },
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x0001",
+        "vid": "0x7472"
+    },
+    "layouts": {
+        "LAYOUT_4x2": {
```

The convention is `<rows>x<cols>`, but as there is only a single layout (and that layout is not an existing community layout)
```suggestion
        "LAYOUT": {
```